docs/configs-npb-tests/run_npb.py

```diff
@@ -93,42 +98,50 @@ def writeBenchScript(dir, bench):
 
     globalStart = time.time()
 
-
     print("Running the simulation")
+    print("Using cpu: {}".format(cpu))
     exit_event = m5.simulate()
 
-    # While there is still something to do in the guest keep executing.
-    # This is needed since we exit for the ROI begin/end
-    foundROI = False
-    while exit_event.getCause() != "m5_exit instruction encountered":
-        # If the user pressed ctrl-c on the host, then we really should exit
-        if exit_event.getCause() == "user interrupt received":
-            print("User interrupt. Exiting")
-            break
+    if exit_event.getCause() == "m5_exit instruction encountered":
+        # Reached the start of ROI
+        # start of ROI is marked by an
+        # m5_exit() call
+        print("Resetting stats at the start of ROI!")
+        m5.stats.reset()
+        start_tick = m5.curTick()
+        start_insts = system.totalInsts()
+        # switching to atomic cpu if argument cpu == atomic
+        if cpu == 'atomic':
+            system.switchCpus(system.cpu, system.atomicCpu)
+    else:
+        print("Unexpected termination of simulation !")
+        exit()
+
+    # Simulate the ROI
+    exit_event = m5.simulate()
 
-        print("Exited because", exit_event.getCause())
+    # Reached the end of ROI
+    # Finish executing the benchmark with kvm cpu
 
-        if exit_event.getCause() == "a thread reached the max instruction count":
-          break
+    print("Dump stats at the end of the ROI!")
+    m5.stats.dump()
+    end_tick = m5.curTick()
+    end_insts = system.totalInsts()
+    m5.stats.reset()
 
-        if exit_event.getCause() == "work started count reach":
-            start_tick = m5.curTick()
-            start_insts = system.totalInsts()
-            foundROI = True
-        elif exit_event.getCause() == "work items exit count reached":
-            end_tick = m5.curTick()
-            end_insts = system.totalInsts()
+    # switch cpu back to kvm if atomic was used for ROI
+    if cpu == 'atomic':
+        system.switchCpus(system.atomicCpu, system.cpu)
 
-        print("Continuing")
-        exit_event = m5.simulate()
+    # Simulate the remaning part of the benchmark
+    exit_event = m5.simulate()
 
+    print("Done with the simulation")
     print()
-    print("Performance statistics")
+    print("Performance statistics:")
 
+    print("Simulated time in ROI: %.2fs" % ((end_tick-start_tick)/1e12))
+    print("Instructions executed in ROI: %d" % ((end_insts-start_insts)))
     print("Ran a total of", m5.curTick()/1e12, "simulated seconds")
     print("Total wallclock time: %.2fs, %.2f min" % \
-                (time.time()-globalStart, (time.time()-globalStart)/60))
-
-    if foundROI:
-        print("Simulated time in ROI: %.2fs" % ((end_tick-start_tick)/1e12))
-        print("Instructions executed in ROI: %d" % ((end_insts-start_insts)))
+                (time.time()-globalStart, (time.time()-globalStart)/60))
```

Comment:
We lost the newline at the end of the file

Reply:
fixed in a new commit